### PR TITLE
chore: Updating Azure CNI and NPM to v1.0.30

### PR DESCRIPTION
**Pull request by jaer-tsun**
**Reason for Change**:

Updates to the latest release of Azure CNI and Azure NPM:
- removing telemetry sidecar from NPM
- set NPM resource limits

https://github.com/Azure/azure-container-networking/releases/tag/v1.0.30

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: packer/install-dependencies.sh [hunk 1]
```diff
@@ -62,8 +62,8 @@ installGPUDrivers
 echo "  - nvidia-docker2 nvidia-container-runtime" >> ${VHD_LOGS_FILEPATH}
 
 VNET_CNI_VERSIONS="
+1.0.30
 1.0.29
-1.0.28
```
Comment (jackfrancis) [inline packer/install-dependencies.sh]: Let's keep 1.0.28 in the VHD as well to allow folks to choose that version, and for easy emergency rollback w/ VHD backing

File: packer/install-dependencies.sh [hunk 2]
```diff
@@ -307,25 +307,15 @@ for AZURE_CNI_NETWORKMONITOR_VERSION in ${AZURE_CNI_NETWORKMONITOR_VERSIONS}; do
 done
 
 AZURE_NPM_VERSIONS="
+1.0.30
 1.0.29
-1.0.28
```
Comment (jackfrancis) [inline packer/install-dependencies.sh]: ditto let's keep 1.0.28

File: packer/install-dependencies.sh [hunk 3]
```diff
@@ -307,25 +307,15 @@ for AZURE_CNI_NETWORKMONITOR_VERSION in ${AZURE_CNI_NETWORKMONITOR_VERSIONS}; do
...
 "
 for AZURE_NPM_VERSION in ${AZURE_NPM_VERSIONS}; do
     CONTAINER_IMAGE="${AZURE_CNIIMAGEBASE}/azure-npm:v${AZURE_NPM_VERSION}"
     pullContainerImage "docker" ${CONTAINER_IMAGE}
     echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
 done
 
-AZURE_VNET_TELEMETRY_VERSIONS="
```
Comment (jackfrancis) [inline packer/install-dependencies.sh]: Do 1.0.29 and 1.0.28 work without the sidecar? If not, let's keep this in the VHD spec as well.

File: parts/k8s/containeraddons/kubernetesmasteraddons-azure-npm-daemonset.yaml [hunk 4]
```diff
@@ -79,6 +79,11 @@ spec:
           image: {{ContainerImage "azure-npm-daemonset"}}
           securityContext:
             privileged: true
+          resources:
+            requests:
+              memory: 20Mi
```
Comment (jackfrancis) [inline parts/k8s/containeraddons/kubernetesmasteraddons-azure-npm-daemonset.yaml]: Let's make this configurable, and set "20mi" as the default. In the yaml spec here you would put:

`{{ContainerMemReqs "azure-npm-daemonset"}}` in place of the static `20mi`

File: parts/k8s/containeraddons/kubernetesmasteraddons-azure-npm-daemonset.yaml [hunk 5]
```diff
@@ -79,6 +79,11 @@ spec:
           image: {{ContainerImage "azure-npm-daemonset"}}
           securityContext:
             privileged: true
+          resources:
+            requests:
+              memory: 20Mi
+            limits:
+              memory: 2Gi
```
Comment (jackfrancis) [inline parts/k8s/containeraddons/kubernetesmasteraddons-azure-npm-daemonset.yaml]: And here replace the static `2Gi` with `{{ContainerMemLimits "azure-npm-daemonset"}}`

File: packer/install-dependencies.sh (hunk 3 above)
Comment (jaer-tsun) [reply packer/install-dependencies.sh]: No, they won't work w/o sidecar. Should I revert the changes in parts/k8s/containeraddons/kubernetesmasteraddons-azure-npm-daemonset.yaml as well?

File: pkg/api/addons.go [hunk 6]
```diff
@@ -322,15 +322,11 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		Containers: []KubernetesContainerSpec{
 			{
 				Name:  common.AzureNetworkPolicyAddonName,
-				Image: "mcr.microsoft.com/containernetworking/azure-npm:v1.0.29",
+				Image: "mcr.microsoft.com/containernetworking/azure-npm:v1.0.30",
```
Comment (jackfrancis) [inline pkg/api/addons.go]: After this line we will add the following 2 container-specific configuration values to enable memory resource limits and requests:

```
MemoryRequests: "20Mi",
MemoryLimits:   "2Gi",
```

**Comment by mboersma** [comment]
@jaer-tsun I mirrored the [linux](https://acs-mirror.azureedge.net/cni/azure-vnet-cni-linux-amd64-v1.0.30.tgz) and [windows](https://acs-mirror.azureedge.net/cni/azure-vnet-cni-windows-amd64-v1.0.30.zip) CNI archives to our production blob store, so this should be good to test.

File: pkg/api/addons_test.go [hunk 7]
```diff
@@ -4311,11 +4311,7 @@ func TestSetAddonsConfig(t *testing.T) {
 					Containers: []KubernetesContainerSpec{
 						{
 							Name:  common.AzureNetworkPolicyAddonName,
-							Image: "mcr.microsoft.com/containernetworking/azure-npm:v1.0.29",
-						},
-						{
-							Name:  common.AzureVnetTelemetryContainerName,
-							Image: "mcr.microsoft.com/containernetworking/azure-vnet-telemetry:v1.0.29",
+							Image: "mcr.microsoft.com/containernetworking/azure-npm:v1.0.30",
```
Comment (jackfrancis) [inline pkg/api/addons_test.go]: You will also have to update each UT case for which Azure NPM is enabled to expect the new `MemoryRequests` and `MemoryLimits` keys and values set as defaults.

File: packer/install-dependencies.sh (hunk 3 above)
Comment (jackfrancis) [reply packer/install-dependencies.sh]: Got it. We can add a conditional go template expression to the yaml spec "if < 1.0.30 then include these extra pods" thing.

Let's not worry about that now, we can iterate there. In the meantime for convenience let's keep everything in the VHD that was already there.

File: packer/install-dependencies.sh (hunk 1 above)
Comment (jaer-tsun) [reply packer/install-dependencies.sh]: added it back

File: packer/install-dependencies.sh (hunk 2 above)
Comment (jaer-tsun) [reply packer/install-dependencies.sh]: added it back

File: pkg/api/addons_test.go (hunk 7 above)
Comment (jaer-tsun) [reply pkg/api/addons_test.go]: updated each UT w/ those

File: pkg/api/addons.go (hunk 6 above)
Comment (jaer-tsun) [reply pkg/api/addons.go]: added

File: parts/k8s/containeraddons/kubernetesmasteraddons-azure-npm-daemonset.yaml (hunk 5 above)
Comment (jaer-tsun) [reply parts/k8s/containeraddons/kubernetesmasteraddons-azure-npm-daemonset.yaml]: fixed

File: parts/k8s/containeraddons/kubernetesmasteraddons-azure-npm-daemonset.yaml (hunk 4 above)
Comment (jaer-tsun) [reply parts/k8s/containeraddons/kubernetesmasteraddons-azure-npm-daemonset.yaml]: fixed

**Pull request closed without merging**